src/validation.cpp

```diff
@@ -935,9 +936,17 @@ bool ReadBlockFromDisk(CBlock& block, const FlatFilePos& pos, const Consensus::P
         return error("%s: Deserialize or I/O error - %s at %s", __func__, e.what(), pos.ToString());
     }
 
-    // Check the header
-    if (!CheckProofOfWork(block.GetHash(), block.nBits, consensusParams))
-        return error("ReadBlockFromDisk: Errors in block header at %s", pos.ToString());
+    // We skip POW checks for genesis block with signet
+    if (!g_signet_blocks || block.GetHash() != consensusParams.hashGenesisBlock) {
```

Comment:
just unconditionally skip PoW checks for genesis block, this block is overly confusing :)

Reply:
Trying to avoid knee-jerk as this will run on mainnet as well. Maybe you're right tho.

Reply:
I think it's harder to read and be sure it's not messing with bitcoin logic. YMMV